### PR TITLE
Implementation of representations of Lie algebras

### PR DESCRIPTION
**Pull request by tscrim**
We provide an implementation of two classes of representations of Lie algebras: the trivial representation and those defined by explicitly giving the matrices.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by tscrim** [comment]
@adityadwarkesh This will be used for the Chevalley-Eilenberg complex in#36049.

**Comment by fchapoton** [comment]
incomplete sentence:

Return whether self is a left or right.

**Comment by tscrim** [comment]
> incomplete sentence:
> 
> Return whether self is a left or right.

Good catch; fixed. I made a few other small tweaks too to make the doc reflect the behavior.

File: src/sage/categories/finite_dimensional_lie_algebras_with_basis.py [hunk 1]
```diff
@@ -1532,6 +1532,37 @@ def morphism(self, on_generators, codomain=None, base_map=None, check=True):
...
+                sage: L.<x,y> = LieAlgebra(QQ, {('x','y'): {'y':1}})
+                sage: f = {x: Matrix([[1,0],[0,0]]), y: Matrix([[0,1],[0,0]])}
+                sage: L.representation(f)
+                Representation of Lie algebra on 2 generators (x, y) over Rational Field defined by:
+                       [1 0]
+                x |--> [0 0]
+                       [0 1]
+                y |--> [0 0]
+            """
+            if f is None:
```
Comment (fchapoton) [inline src/sage/categories/finite_dimensional_lie_algebras_with_basis.py]: needs a doctest for the trivial repr here above

**Comment by fchapoton** [comment]
some raise statements needs corresponding doctests, as asked by CodeCoverage

**Comment by tscrim** [comment]
I added the tests. Although I realized I could easily extend this code to work for finite dimensional representations of any Lie algebra (not just finite dimensional with a basis).

File: src/sage/algebras/lie_algebras/lie_algebra_element.pyx [hunk 2]
```diff
@@ -1232,7 +1232,7 @@ cdef class UntwistedAffineLieAlgebraElement(Element):
         return type(self)(self._parent, negate(self._t_dict),
                           -self._c_coeff, -self._d_coeff)
 
-    cpdef _acted_upon_(self, x, bint self_on_left) noexcept:
+    cpdef _acted_upon_(self, scalar, bint self_on_left) noexcept:
         """
         Return ``self`` acted upon by ``x``.
```
Comment (fchapoton) [inline src/sage/algebras/lie_algebras/lie_algebra_element.pyx:1238]: change x to scalar in the doc please

File: src/sage/categories/lie_algebras.py [hunk 3]
```diff
@@ -697,6 +697,66 @@ def lie_group(self, name='G', **kwds):
...
+            return TrivialRepresentation(self)
+
+        def representation(self, f=None, index_set=None, on_basis=None, **kwargs):
+            """
+            Return a representation of ``self``.
+
+            If no arguments are given, then this returns the trivial
+            representation.
+
+            Currently the only implementated method of constructing a
```
Comment (fchapoton) [inline src/sage/categories/lie_algebras.py]: typo "implementated"

File: src/sage/categories/lie_algebras.py [hunk 4]
```diff
@@ -697,6 +697,66 @@ def lie_group(self, name='G', **kwds):
...
+                       [1 0]
+                x |--> [0 0]
+                       [0 1]
+                y |--> [0 0]
+                sage: L.representation()
+                Trivial representation of Lie algebra on 2 generators (x, y) over Rational Field
+            """
+            if f is None and on_basis is None and index_set is None:
+                return self.trivial_representation(**kwargs)
+            if on_basis is None:
```
Comment (fchapoton) [inline src/sage/categories/lie_algebras.py]: is it common to use `on_basis=None` as default ? why not use `on_basis=False` as default ?

Comment (tscrim) [reply src/sage/categories/lie_algebras.py]: Good point. Changed.

**Comment by tscrim** [comment]
Okay, added the more tests. Although I am still not 100% convinced all of these tests that the safety checkers are raising an error is useful as we are indirectly testing it by the code not falling into it and it is bad user input.

**Comment by fchapoton** [comment]
ok, let's move this in

**Review by fchapoton — APPROVED** [review]
ok, good to go

**Comment by tscrim** [comment]
Thank you!

**Pull request merged**